test/e2e/storage/utils/deployment.go

```diff
@@ -93,6 +93,10 @@ func PatchCSIDeployment(f *framework.Framework, o PatchCSIOptions, object interf
 				// Driver name is expected to be the same
 				// as the provisioner here.
 				container.Args = append(container.Args, "--provisioner="+o.NewDriverName)
+			case o.SnapshotterContainerName:
+				// Driver name is expected to be the same
+				// as the snapshotter here.
+				container.Args = append(container.Args, "--snapshotter="+o.NewDriverName)
```

Comment:
Let's fix this in a separate PR, but we should remove the snapshotter argument from external-snapshotter since it must always be equal to the driver name.  It was a mistake to have this in the external provisioner and there is a bug open to remove it.

Reply:
When we run the e2e every time, the driver name is different 
```
NewDriverName:     "csi-hostpath-" + h.f.UniqueName,
``` 
so we should clean the test logic too, I agree that we can fix it in a separate pr.